src/main/java/org/amahi/anywhere/activity/AuthenticationActivity.java

```diff
@@ -153,161 +191,174 @@ private void setUpAuthenticationActionListener() {
 
 	@Override
 	public void onClick(View view) {
-		if(getUsername().trim().isEmpty() || getPassword().trim().isEmpty()){
-			ViewDirector.of(this,R.id.animator_message).show(R.id.text_message_authentication_empty);
+      getUsernameEdit().clearFocus();
+      getPasswordEdit().clearFocus();
+    
+      if (getUsername().trim().isEmpty() || getPassword().trim().isEmpty()) {
+            imageView.setImageResource(R.drawable.ic_banner);
 
-			if(getUsername().trim().isEmpty())
-				getUsernameEdit().getBackground().setColorFilter(ContextCompat.getColor(AuthenticationActivity.this, android.R.color.holo_red_light), PorterDuff.Mode.SRC_ATOP);
-			if(getPassword().trim().isEmpty())
-				getPasswordEdit().getBackground().setColorFilter(ContextCompat.getColor(AuthenticationActivity.this, android.R.color.holo_red_light), PorterDuff.Mode.SRC_ATOP);
+            InputMethodManager imm = (InputMethodManager) getSystemService(Context.INPUT_METHOD_SERVICE);
+            imm.hideSoftInputFromWindow(getCurrentFocus().getWindowToken(), InputMethodManager.RESULT_UNCHANGED_SHOWN);
 
-			getUsernameEdit().addTextChangedListener(new TextWatcher() {
-				@Override
-				public void beforeTextChanged(CharSequence charSequence, int i, int i1, int i2) {
+            ViewDirector.of(this, R.id.animator_message).show(R.id.text_message_authentication_empty);
 
-				}
+            if (getUsername().trim().isEmpty())
+                getUsernameEdit().getBackground().setColorFilter(getResources().getColor(android.R.color.holo_red_light), PorterDuff.Mode.SRC_ATOP);
```

Comment:
On resolving the merge conflicts you have changed back the `ContextCompat.getColor` line to the `getResouces().getColor` which is deprecated and was previously resolved.